src/node_url.cc

```diff
@@ -1698,7 +1710,8 @@ void URL::Parse(const char* input,
         } else {
           if (has_base &&
               base->scheme == "file:") {
-            if (IsNormalizedWindowsDriveLetter(base->path[0])) {
+            if (IsNormalizedWindowsDriveLetter(base->path[0]) &&
+              !StartsWithWindowsDriveLetter(ch, p, remaining)) {
```

Comment:
@rmisev thank you for the through review, I've implemented your suggestions. I think it might be good to eventually pull in all of `url/urltestdata.json` to our test suite, I might do that as a follow up pull request (thought @TimothyGu?).